core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,243 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+constexpr static double aInit  = 1.0;
+constexpr static double bInit  = 2.0;
+constexpr static double cInit  = 3.0;
+constexpr static double scalar = 4.0;
+
+using StreamDeviceArray =
+    Kokkos::View<double*, Kokkos::MemoryTraits<Kokkos::Restrict>>;
```

Comment:
It would be interesting to add a comment about the choice of using the `restrict` keyword. Have you had a chance to evaluate its impact on the results?

Reply:
I have added benchmarks with and without restrict. For CUDA on my workstation it didn't make a difference, but it helped the OpenMP backend substantially (~15%).